### ISSUE TITLE
JSDoc validation with @param

### ISSUE THREAD

**Issue by angelozerr** [angelozerr/tern-lint#42]
Support this JSDoc validation :

```javascript
/**
 * @param {number} millis
 * @param {function} callback
 * */
function sleep(millis, callback) {
    setTimeout(function() {
        callback();
    }, millis);
}

sleep('aaa', 'sdsdsd'); //  'aaa' and 'sdsdsd' must raise an error
```

This sample comes from http://stackoverflow.com/questions/22218964/jsdoc-how-to-check-types-for-javascript
